src/ui/ui_exports.js

```diff
@@ -37,22 +43,43 @@ class UiExports {
     this.consumers.push(consumer);
   }
 
+  // when module ids are defined in a uiExports spec they can
+  // be defined as relative to the public directory
+  resolveModulePath(plugin, moduleId) {
+    const isRelative = moduleId.startsWith('./') || moduleId.startsWith('.\\');
+    if (isRelative) return join(`plugins/${plugin.id}`, moduleId);
+    return moduleId;
+  }
+
+  extendAliases(plugin, alias, idOrIds) {
+    const existingModuleIds = this.aliases[alias] || [];
+    const resolve = id => this.resolveModulePath(plugin, id);
+    const resolvedModuleIds = arr(idOrIds).map(resolve);
+
+    this.aliases[alias] = _.union(existingModuleIds, resolvedModuleIds);
+  }
+
   exportConsumer(type) {
-    for (let consumer of this.consumers) {
+    for (const consumer of this.consumers) {
       if (!consumer.exportConsumer) continue;
-      let fn = consumer.exportConsumer(type);
+      const fn = consumer.exportConsumer(type);
       if (fn) return fn;
     }
 
     switch (type) {
-      case 'app':
       case 'apps':
-        return (plugin, specs) => {
-          for (let spec of [].concat(specs || [])) {
-            let app = this.apps.new(_.defaults({}, spec, {
+        return (plugin, spec) => {
+          for (const rawSpec of arr(spec)) {
```

Comment:
I see, that makes sense.

This kind of underscores the fundamental issue behind the point I brought up about `visTypes`, but it's relevant here: when we allow for multiple ways to configure the exact same thing, consuming code just becomes more complicated because we have to be defensive about things like that.

And in this particular case, people may not have any idea that they can export multiple apps from a plugin because most plugin examples they see use the single app convention.

I don't believe these configuration features that we've added for the sake of convenience are ultimately worth the complexity cost.